### PR TITLE
Fix #3570 'Box data left in ~/.vagrant.d/boxes after removal'

### LINKED ISSUE #3570: Box data left in `~/.vagrant.d/boxes` after removal
I removed two boxes (```alpine-2.7.5-amd64``` and ```ubuntu-14.04-amd64```), but after that there is still data left in ```.vagrant.d/boxes```. Should be removed, too.

```
% vagrant box list
berendt/devstack-ubuntu-14.04-amd64 (virtualbox, 0.1.5)
```

```
% ll .vagrant.d/boxes 
total 12
drwxr-xr-x 3 berendt users 4096 Mar 31 20:11 berendt-VAGRANTSLASH-alpine-2.7.5-amd64/
drwxr-xr-x 3 berendt users 4096 Apr 29 10:04 berendt-VAGRANTSLASH-devstack-ubuntu-14.04-amd64/
drwxr-xr-x 3 berendt users 4096 Apr  6 18:56 berendt-VAGRANTSLASH-ubuntu-14.04-amd64/
```

### PR DESCRIPTION
**Pull request by ievgenp**
As per my comment at the issue page, please check if I've missed some sort of concurrency check in my patch.

### CODE REVIEW

**Comment by ievgenp** [comment]
I was talking about my comment in this issue #3570. Mentioning it here for easy access to its page.

**Comment by mitchellh** [comment]
Thanks! Sorry I took so long to respond on this one, you opened this PR right as we did Vagrant 1.7.4 and then we left Vagrant issues for awhile. I'm taking a look now but this looks to be mostly there. Instead of doing a standard review cycle I'm going to merge, fix it up, and then I can CC you on changes.

**Comment by mitchellh** [comment]
I'll CC you on changes (but its merged). Thanks!

File: lib/vagrant/box_collection.rb [hunk 1]
```diff
@@ -13,6 +13,8 @@ module Vagrant
   # boxes.
   class BoxCollection
     TEMP_PREFIX = "vagrant-box-add-temp-"
+    VAGRANT_SLASH = "-VAGRANTSLASH-"
+    VAGRANT_COLON = "-VAGRANTCOLON-"
```
Comment (mitchellh) [inline lib/vagrant/box_collection.rb]: Love this, thanks!

**Comment by mitchellh** [comment]
See above commit ^

**Comment by ievgenp** [comment]
Cool! Thank you. Now I'm gonna find another issue for myself.

**Pull request merged**